fem/gridfunc.cpp

```diff
@@ -328,7 +328,28 @@ int GridFunction::VectorDim() const
    {
       return fes->GetVDim();
    }
-   return fes->GetVDim()*fes->GetMesh()->SpaceDimension();
+   return fes->GetVDim()*fe->GetVDim();
+}
+
+int GridFunction::CurlDim() const
+{
+   const FiniteElement *fe;
+   if (!fes->GetNE())
+   {
+      const FiniteElementCollection *fec = fes->FEColl();
+      static const Geometry::Type geoms[3] =
+      { Geometry::SEGMENT, Geometry::TRIANGLE, Geometry::TETRAHEDRON };
+      fe = fec->FiniteElementForGeometry(geoms[fes->GetMesh()->Dimension()-1]);
+   }
+   else
+   {
+      fe = fes->GetFE(0);
+   }
+   if (!fe || fe->GetRangeType() == FiniteElement::SCALAR)
+   {
+      return 2 * fes->GetMesh()->SpaceDimension() - 3;
```

Comment:
Is this correct in 1D, returning -1?

Reply:
I'm not sure it needs to be correct in 1D but I could be mistaken. This `if` statements handles the case where we have multiple copies of a scalar field and our current assumptions are that such a field is either 2 or 3 dimensional. I believe a 1D scalar field, with vdim=1, would throw an error before calling this method.

Of course, it wouldn't hurt to simplify this logic and maybe even document this method so that we know what to expect from it...